indexd/utils.py

```diff
@@ -9,7 +9,17 @@
 
 logger = logging.getLogger(__name__)
 
-
+def handle_error(resp):
+    if 400 <= resp.status_code < 600:
+        try:
+            json = resp.json()
+            resp.reason = json.get("error")
```

Comment:
We don't import the json library in this module but we should be careful not to clobber python keywords and libraries.